components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -111,18 +143,32 @@ export function WorkspaceEntry({ desc, model, isAdmin, stopWorkspace }: Props) {
             </Tooltip>
         </ItemField>
         <ItemFieldContextMenu menuEntries={menuEntries} />
-        {isModalVisible && <ConfirmationModal
+        {isDeleteModalVisible && <ConfirmationModal
             title="Delete Workspace"
             areYouSureText="Are you sure you want to delete this workspace?"
             children={{
                 name: ws.id,
                 description: ws.description,
             }}
             buttonText="Delete Workspace"
-            visible={isModalVisible}
-            onClose={() => setModalVisible(false)}
+            visible={isDeleteModalVisible}
+            onClose={() => setDeleteModalVisible(false)}
             onConfirm={() => model.deleteWorkspace(ws.id)}
         />}
+        <Modal visible={isRenameModalVisible} onClose={() => setRenameModalVisible(false)} onEnter={() => { updateWorkspaceDescription(); return true }}>
+            <h3 className="mb-4">Rename Workspace Description</h3>
+            <div className="border-t border-b border-gray-200 dark:border-gray-800 -mx-6 px-6 py-4 space-y-2">
+                <input className="w-full" type="text" defaultValue={workspaceDescription} ref={renameInputRef} />
+                <div className="mt-1">
+                    <p className="text-gray-500">You can use a friendlier name for this workspace description.</p>
```

Comment:
**nitpick**: Re-read this and thought a better alternative could be something like the following. What do you think? This will overflow. 💡 

```suggestion
                    <p className="text-gray-500">Change the workspace description to make it easier to go back to a workspace.</p>
```